runtime/stdlib/test_emulatorbackend.go

```diff
@@ -494,6 +533,129 @@ func (t *testEmulatorBackendType) newUseConfigFunction(testFramework TestFramewo
+
+const testEmulatorBackendTypeLogsFunctionDocString = `
+Returns all the logs from the blockchain, up to the calling point.
+`
+
+func (t *testEmulatorBackendType) newLogsFunction(
+	testFramework TestFramework,
+) *interpreter.HostFunctionValue {
+	return interpreter.NewUnmeteredHostFunctionValue(
+		t.logsFunctionType,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			logs := testFramework.Logs()
+			inter := invocation.Interpreter
+
+			arrayType := interpreter.NewVariableSizedStaticType(
+				inter,
+				interpreter.NewPrimitiveStaticType(
+					inter,
+					interpreter.PrimitiveStaticTypeString,
+				),
+			)
+
+			values := make([]interpreter.Value, len(logs))
+			for i, log := range logs {
+				memoryUsage := common.NewStringMemoryUsage(len(log))
+				values[i] = interpreter.NewStringValue(
+					inter,
+					memoryUsage,
+					func() string {
+						return log
+					},
+				)
+			}
+
+			return interpreter.NewArrayValue(
+				inter,
+				invocation.LocationRange,
+				arrayType,
+				common.ZeroAddress,
+				values...,
+			)
+		},
+	)
+}
+
+// 'EmulatorBackend.serviceAccount' function
+
+const testEmulatorBackendTypeServiceAccountFunctionName = "serviceAccount"
+
+const testEmulatorBackendTypeServiceAccountFunctionDocString = `
+Returns the service account of the blockchain. Can be used to sign
+transactions with this account.
+`
+
+func (t *testEmulatorBackendType) newServiceAccountFunction(
+	testFramework TestFramework,
+) *interpreter.HostFunctionValue {
+	return interpreter.NewUnmeteredHostFunctionValue(
+		t.serviceAccountFunctionType,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			serviceAccount, err := testFramework.ServiceAccount()
+			if err != nil {
+				panic(err)
+			}
+
+			return newTestAccountValue(
+				testFramework,
+				invocation.Interpreter,
+				invocation.LocationRange,
+				serviceAccount,
+			)
+		},
+	)
+}
+
+// 'EmulatorBackend.getEvents' function
+
+const testEmulatorBackendTypeGetEventsFunctionName = "getEvents"
+
+const testEmulatorBackendTypeGetEventsFunctionDocString = `
+Returns all events emitted from the blockchain,
+optionally filtered by event type.
+`
+
+func (t *testEmulatorBackendType) newGetEventsFunction(
+	testFramework TestFramework,
+) *interpreter.HostFunctionValue {
+	return interpreter.NewUnmeteredHostFunctionValue(
+		t.getEventsFunctionType,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			eventType, ok := invocation.Arguments[0].(*interpreter.StringValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+			events := testFramework.GetEvents(eventType.Str)
+
+			inter := invocation.Interpreter
+
+			arrayType := interpreter.NewVariableSizedStaticType(
+				inter,
+				interpreter.PrimitiveStaticTypeAnyStruct,
+			)
+
+			values := make([]interpreter.Value, len(events))
+			// for i, event := range events {
+			//  How do I convert a cadence.Event to interpreter.Value ?
+			// 	values[i] = event
+			// }
```

Comment:
Conversion should be possible using `runtime.ImportValue`

Reply:
Perfect, I found the relevant comments also: https://github.com/onflow/cadence/blob/master/runtime/interpreter/value.go#L102-L103. However, I think this should be done from the `EmulatorBackend` implementation, I ran into some cyclic import errors, while trying to use `runtime.ImportValue` inside the `stdlib` package (which already import `runtime`). Like the `RunScript` implementation https://github.com/onflow/cadence-tools/blob/8ebd4b9835c4ad46bd28ece858292e367ece137a/test/emulator_backend.go#L167-L176

Reply:
Updated in https://github.com/onflow/cadence-tools/pull/113/commits/72eee50a3699d94abcc474141d6eec48668dfbcc